packages/vitest/src/api/setup.ts

```diff
@@ -114,6 +114,9 @@ export function setup(vitestOrWorkspace: Vitest | WorkspaceProject, _server?: Vi
           await ctx.rerunFiles(files)
         },
         getConfig() {
+          if (vitestOrWorkspace instanceof WorkspaceProject)
+            return wrapConfig(vitestOrWorkspace.getSerializableConfig())
```

Comment:
For now, I added this only for browser mode testing (but not for vitest ui client) since vitest ui doesn't seem to make use of `testNamePattern`.

Alternatively, we could fix this in serialization layer, but it didn't feel appropriate to put this custom logic in `@vitest/ws-client` unless changing serialization library entirely to cover RegExp, Error, etc...

https://github.com/vitest-dev/vitest/blob/76b6195c2b1f241f549752505c5d1a164d8d86dc/packages/ws-client/src/index.ts#L79-L80